### DISCUSSION TITLE
Cross-notes content control system ?

### DISCUSSION THREAD

**Discussion by Nick222** [zim-desktop-wiki/zim-desktop-wiki#2055] [Ideas]
When I working with a Zim-notebook, often it need to move large pieces of text between different notes.
Sometimes I cut text from the note and forget to paste this text to another place - and it gets lost.

Questions:

1) Is there a plugin for Zim (like a clipboard) that would delete text from the clipboard when it was pasted to a new location and would not allow copying or cutting out new text while the old one is still in the clipboard?

2) Is there a tool for comparing the state of a notebook before and after editing - to identify cut and not pasted pieces of text?
The version control system Bazaar too hard for me - because multiple cutting and pasting large pieces of text in different files.

**Comment by krisztianmukli** [comment]
I think you should use a clipboard manager. It is not working how you
write, but I as far as I can see, it can solve your problems... On
Windows 10 try WIN+V keys, or on Linux I'm using Parcellite (with
SHIFT+CTRL+H) or find an other similar software. 

---
KRISZTIÁN MUKLI 

_mail: krisztianmukli@mukli.hu_
_web: www.mukli.hu [3]_ 

08.01.2020 12:40 időpontban Nick222 ezt írta: 

> When I working with a Zim-notebook, often it need to move large pieces of text between different notes.
> Sometimes I cut text from the note and forget to paste this text to another place - and it gets lost. 
> 
> Questions: 
> 
> * 
> 
> Is there a plugin for Zim (like a clipboard) that would delete text from the clipboard when it was pasted to a new location and would not allow copying or cutting out new text while the old one is still in the clipboard? 
> * 
> 
> Is there a tool for comparing the state of a notebook before and after editing - to identify cut and not pasted pieces of text?
> The version control system too hard for me - because multiple cutting and pasting large pieces of text in different files. 
> 
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].
 

Links:
------
[1]
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/997?email_source=notifications&amp;email_token=AAKSS6HNUHNKVUCGX6CPKM3Q4YFXZA5CNFSM4KEMFGM2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IE2WKZA
[2]
https://github.com/notifications/unsubscribe-auth/AAKSS6GVXZJRWB6KUNUZRZ3Q4YFXZANCNFSM4KEMFGMQ
[3] http://www.mukli.hu

**Comment by Nick222** [comment]
After a lengthy editing session, I copied a listing of the changes from the version control system.
Then, in a text editor, I compared the lines with minuses and the lines with pluses - so that for each line with a minus there was the same line with a plus...

Of course, it would be nice to automate such a comparison.

**Comment by Self-Perfection** [comment]
@Nick222 with unix way you can easily script such comparison. Basically you need to get list of all lines now and at some point backwards in time. Here is solution for you:

```bash
# Print all notes except header lines removing empty lines in the process
# and sort this list, so you won't get extra differences if notes would be printed
# in another order in second pass
print_lines() {
  find -name '*.txt' -exec sed -n '/^$/,${/^$/d;p}' {} \; | sort
}

cd "NOTEBOOK_PATH"
print_lines >/tmp/lines_new

# Now you need to repeat this for old notebook state.
# Don't know how to do it with bazaar. I urge you to switch to different VCS
# as bazaar is not developed anymore. With git this is easy:
git worktree add /tmp/Notes_old 'master@{3.hour.ago}'

cd /tmp/Notes_old
print_lines > /tmp/lines_old

cd -
git worktree remove /tmp/Notes_old

diff /tmp/lines_old /tmp/lines_new
```

**Comment by Nick222** [comment]
Move to Discussion, pls...
